test/test_dev.py

```diff
@@ -50,16 +48,6 @@ def generate_basic_conf(tmpdir, repo_subdir=''):
     return tmpdir, local, conf
 
 
-@pytest.fixture
-def basic_conf(tmpdir):
-    return generate_basic_conf(tmpdir)
-
-
-@pytest.fixture
-def basic_conf_with_subdir(tmpdir):
-    return generate_basic_conf(tmpdir, 'some_subdir')
-
-
 def test_dev(capsys, basic_conf):
     tmpdir, local, conf = basic_conf
```

Comment:
@datapythonista Thank you for your comment I created an issue #1130 that has more context on this,

 I also read about it n noticed Unpacking works when the number of variables and the numbers of values is the same. am guessing for this particular case `basic_conf` from conftest.py is returning more variables(return tmpdir, local, conf, machine_file) as seen below.This complains for all functions listed that requires unpacking 
```
=========================== short test summary info ============================
FAILED test/test_check.py::test_check - ValueError: too many values to unpack...
FAILED test/test_dev.py::test_dev - ValueError: too many values to unpack (ex...
FAILED test/test_dev.py::test_dev_with_repo_subdir - ValueError: too many val...
FAILED test/test_dev.py::test_dev_strict - ValueError: too many values to unp...
FAILED test/test_dev.py::test_run_python_same - ValueError: too many values t...
FAILED test/test_dev.py::test_profile_python_same - ValueError: too many valu...
============ 6 failed, 194 passed, 16 skipped in 644.90s (0:10:44) =============
```
below is the conftest.py `basic_conf` function being referenced

```
@pytest.fixture
def basic_conf(tmpdir, dummy_packages):
    return generate_basic_conf(tmpdir)

```

and the generate_basic_conf function below;

```
def generate_basic_conf(tmpdir, repo_subdir='', values=dummy_values, dummy_packages=True):
    tmpdir = str(tmpdir)
    local = abspath(dirname(__file__))
    os.chdir(tmpdir)

    # Use relative paths on purpose since this is what will be in
    # actual config files

    shutil.copytree(os.path.join(local, 'benchmark'), 'benchmark')

    machine_file = join(tmpdir, 'asv-machine.json')

    shutil.copyfile(join(local, 'asv-machine.json'),
                    machine_file)

    repo_path = tools.generate_test_repo(tmpdir, values,
                                         subdir=repo_subdir).path

    if dummy_packages:
        matrix = {
            "asv_dummy_test_package_1": [""],
            "asv_dummy_test_package_2": tools.DUMMY2_VERSIONS,
        }
    else:
        matrix = {}

    conf_dict = {
        'env_dir': 'env',
        'benchmark_dir': 'benchmark',
        'results_dir': 'results_workflow',
        'html_dir': 'html',
        'repo': relpath(repo_path),
        'dvcs': 'git',
        'project': 'asv',
        'matrix': matrix,
    }
    if repo_subdir:
        conf_dict['repo_subdir'] = repo_subdir

    conf = config.Config.from_json(conf_dict)

    if hasattr(sys, 'pypy_version_info'):
        conf.pythons = ["pypy{0[0]}.{0[1]}".format(sys.version_info)]

    return tmpdir, local, conf, machine_file
```

Reply:
Unpacking happens when you assign to a tuple, instead of a scalar, and the number of elements assigned and being assigned need to match. But please, instead of copying code, add a github comment to the relevant lines in the diff. And mention in the comment about the exact problem you're having. I'm not going to fix this error for you, and do the research on what's failing. That's your job. I'm happy to help you unblock in the specific part you can't understand yourself, but for that I need to know what it is.

Reply:
Thank you @datapythonista